src/dvui.zig

```diff
@@ -1102,19 +1102,125 @@ pub const TextureTarget = struct {
 pub const TextureCacheEntry = struct {
     texture: Texture,
 
-    pub fn hash(bytes: []const u8, height: u32) u64 {
+    fn hash(bytes: []const u8, height: u32) u64 {
         var h = fnv.init();
         h.update(std.mem.asBytes(&bytes.ptr));
         h.update(std.mem.asBytes(&height));
         return h.final();
     }
-    pub fn hash_icon(bytes: []const u8, height: u32, opt: IconRenderOptions) u64 {
+    fn hash_icon(bytes: []const u8, height: u32, opt: IconRenderOptions) u64 {
         var h = fnv.init();
         h.update(std.mem.asBytes(&bytes.ptr));
         h.update(std.mem.asBytes(&height));
         h.update(std.mem.asBytes(&opt));
         return h.final();
     }
+
+    pub fn fromImageFile(name: []const u8, image_bytes: []const u8) (Backend.TextureError || StbImageError)!TextureCacheEntry {
+        var cw = currentWindow();
+        const tex_hash = TextureCacheEntry.hash(image_bytes, 0);
+
+        if (cw.texture_cache.get(tex_hash)) |tce| return tce;
+
+        var w: c_int = undefined;
+        var h: c_int = undefined;
+        var channels_in_file: c_int = undefined;
+        const data = c.stbi_load_from_memory(image_bytes.ptr, @as(c_int, @intCast(image_bytes.len)), &w, &h, &channels_in_file, 4);
+        if (data == null) {
+            log.warn("imageTexture stbi_load error on image \"{s}\": {s}\n", .{ name, c.stbi_failure_reason() });
+            return StbImageError.stbImageError;
+        }
+
+        defer c.stbi_image_free(data);
+
+        var pixels: []u8 = undefined;
+        pixels.ptr = data;
+        pixels.len = @intCast(w * h * 4);
+
+        const texture = try textureCreate(.fromRGBA(pixels), @intCast(w), @intCast(h), .linear);
+
+        const entry = TextureCacheEntry{ .texture = texture };
+        try cw.texture_cache.put(cw.gpa, tex_hash, entry);
+
+        return entry;
+    }
+
+    pub fn fromRgbaPixelPMA(pma: dvui.RGBAPixelsPMA, width: u32, height: u32) !dvui.TextureCacheEntry {
```

Comment:
The type of the first parameter makes it clear which format the pixels should be. Also adding specific error type.

```suggestion
    pub fn fromPixels(pma: dvui.RGBAPixelsPMA, width: u32, height: u32) Backend.TextureError!dvui.TextureCacheEntry {
```

Reply:
theoretically we could support other pixel format, this is why i chose this overly specific name. but either way i am fine. unlikely to support anything else than rgba []u8 and we would probably convert to RGBAPixelsPMA anyway